cachi2/interface/cli.py

```diff
@@ -405,6 +405,94 @@ def all_files_are_jsons(sbom_files_to_merge: Paths) -> Paths:
     return all_files_are_jsons(enough_files_for_merge(list(set(sbom_files_to_merge))))
 
 
+def merge_relationships(
+    relationships_list: List[List[SPDXRelation]], doc_ids: List[str], packages: List[SPDXPackage]
+) -> List[SPDXRelation]:
+    """Merge SPDX relationships.
+
+    Function takes two relationsships lists and unified list of packages.
+    For relationhips lists, map and inverse map of relations is created. SPDX document usually
+    contains virtual package which serves as "envelope" for all real packages. These virtual
+    packages are search in the relationsships and thier ID is stored as middle element.
+    """
+
+    def map_relationships(
+        relationships: List[SPDXRelation],
+    ) -> Tuple[Optional[str], Dict[str, List[str]], Dict[str, str]]:
+        relations_map: Dict[str, List[str]] = {}
+        inverse_map: Dict[str, str] = {}
+
+        for rel in relationships:
+            spdx_id, related_spdx = rel.spdxElementId, rel.relatedSpdxElement
+            relations_map.setdefault(spdx_id, []).append(related_spdx)
+            inverse_map[related_spdx] = spdx_id
+
+        root_element = next((k for k in relations_map if k not in inverse_map), None)
+        return root_element, relations_map, inverse_map
+
+    package_ids = {pkg.SPDXID for pkg in packages}
+    root_ids = []
+    maps = []
+    inv_maps = []
+    middles = []
```

Comment:
I wonder if `envelop_packages_ids` would be better in conveying what this variable refers to? I found myself going back to the docstring several times to refresh what 'middle' and 'middles' mean.

Reply:
Yeah, I can rename it.